feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,275 @@
+	temperatureMinSample := gnmi.Collect(t, gnmiOpts(t, dut1, gpb.SubscriptionMode_SAMPLE, time.Second*10), component1.Temperature().Min().State(), time.Second*10)
+	temperatureMinSamples := temperatureMinSample.Await(t)
+	if len(temperatureMinSamples) == 0 {
+		t.Fatalf("Get Temeprature Min sample list for interface %q: got 0, want > 0", dp1.Name())
+	}
+	// Taking first sample
+	temperatureMin, _ := temperatureMinSamples[0].Val()
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	if reflect.TypeOf(temperatureMin).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	} else if temperatureMin <= 0 && temperatureMin >= 300 {
+		t.Fatalf("The variable temperature min is not between 0 and 300")
+	}
+	temperatureMaxSample := gnmi.Collect(t, gnmiOpts(t, dut1, gpb.SubscriptionMode_SAMPLE, time.Second*10), component1.Temperature().Max().State(), time.Second*10)
+	temperatureMaxSamples := temperatureMaxSample.Await(t)
+	if len(temperatureMaxSamples) == 0 {
+		t.Fatalf("Get Temeprature Max sample list for interface %q: got 0, want > 0", dp1.Name())
+	}
+	// Taking first sample
+	temperatureMax, _ := temperatureMaxSamples[0].Val()
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	if reflect.TypeOf(temperatureMax).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	} else if temperatureMax <= 0 && temperatureMax >= 300 {
+		t.Fatalf("The variable temperature min is not between 0 and 300")
+	}
+	temperatureAvgSample := gnmi.Collect(t, gnmiOpts(t, dut1, gpb.SubscriptionMode_SAMPLE, time.Second*10), component1.Temperature().Avg().State(), time.Second*10)
+	temperatureAvgSamples := temperatureAvgSample.Await(t)
+	if len(temperatureAvgSamples) == 0 {
+		t.Fatalf("Get Temeprature Avg sample list for interface %q: got 0, want > 0", dp1.Name())
+	}
+	// Taking first sample
+	temperatureAvg, _ := temperatureAvgSamples[0].Val()
+	if reflect.TypeOf(temperatureAvg).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	} else if temperatureAvg <= 0 && temperatureAvg >= 300 {
+		t.Fatalf("The variable temperature avg is not between 0 and 300")
+	}
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	// temperature average should be between min and max
+	if temperatureAvg >= temperatureMin && temperatureAvg <= temperatureMax {
+		t.Logf("The average is between the maximum and minimum values")
+	} else {
+		t.Fatalf("The average is not between the maximum and minimum values")
+	}
+	subcomponents := gnmi.LookupAll[*oc.Component_Subcomponent](t, dut1, component1.SubcomponentAny().State())
+	for _, s := range subcomponents {
+		subc, ok := s.Val()
+		if ok {
+			sensorComponent := gnmi.Get[*oc.Component](t, dut1, gnmi.OC().Component(subc.GetName()).State())
+			if sensorComponent.GetType() == sensorType {
+				scomponent := gnmi.OC().Component(sensorComponent.GetName())
```

Comment:
Thanks for the changes, Sunil! 
I'd suggest one further change to this. You don't need to separate lookups for the subcomponent temperature paths and the transceiver temperature. 

I think the logic you want is:
1) lookup the transceiver name (which you already added at https://github.com/openconfig/featureprofiles/blob/3a0e1ed449b17e6e65f8951688b968b9e888461d/feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go#L51) and component path
https://github.com/openconfig/featureprofiles/blob/3a0e1ed449b17e6e65f8951688b968b9e888461d/feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go#L56
2) lookup if there's a temperature `SENSOR` subcomponent for the component, and record its name if its present
3) if p. 2 returned a non-nil sensor subcomponent, perform all temperature lookups on the subcomponent instead of the transceiver component.

I.e. what you want is just to move this lookup piece to the top, after you do L56, and depending on the lookup result (separate temp. sensor can be present or absent) you set the `component1` value as the transceiver path or the sensor path.

Let me know if that makes sense?

Reply:
Hi Sergey,

Updated the script as suggested please review and update.

Thanks